storage/series_test.go

```diff
@@ -119,3 +123,185 @@ func TestChunkSeriesSetToSeriesSet(t *testing.T) {
+		{
+			name: "single histogram to single chunk",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, h: h1},
+			},
+			expectedChunks: 1,
+		},
+		{
+			name: "two histograms encoded to a single chunk",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, h: h1},
+				&sample{t: 2, h: h2},
+			},
+			expectedChunks: 1,
+		},
+		{
+			name: "two histograms encoded to two chunks",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, h: h2},
+				&sample{t: 2, h: h1},
+			},
+			expectedChunks:       2,
+			expectedCounterReset: true,
+		},
+		{
+			name: "histogram and stale sample encoded to two chunks",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, h: staleHistogram},
+				&sample{t: 2, h: h1},
+			},
+			expectedChunks: 2,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Run("histograms", func(t *testing.T) {
+				testHistogramsSeriesToChunks(t, test)
+			})
+			t.Run("float_histograms", func(t *testing.T) {
+				// Convert all histograms to float histograms.
+				for i := range test.samples {
+					test.samples[i].(*sample).fh = test.samples[i].H().ToFloat()
+					test.samples[i].(*sample).h = nil
+				}
+				testHistogramsSeriesToChunks(t, test)
+			})
+		})
+	}
+	t.Run("mixed_histograms", func(t *testing.T) {
+		mixedTests := []histogramTest{{
+			name: "histogram and float histogram encoded to two chunks",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, h: h1},
+				&sample{t: 2, fh: h2.ToFloat()},
+			},
+			expectedChunks: 2,
+		}, {
+			name: "histogram and float histogram encoded to two chunks",
+			lbs:  labels.FromStrings("__name__", "up", "instance", "localhost:8080"),
+			samples: []tsdbutil.Sample{
+				&sample{t: 1, fh: h1.ToFloat()},
+				&sample{t: 2, h: h2},
+			},
+			expectedChunks: 2,
+		}}
+		for _, test := range mixedTests {
+			testHistogramsSeriesToChunks(t, test)
+		}
+	})
+}
+
+func testHistogramsSeriesToChunks(t *testing.T, test histogramTest) {
+	series := NewListSeries(test.lbs, test.samples)
+	encoder := NewSeriesToChunkEncoder(series)
+	require.EqualValues(t, test.lbs, encoder.Labels())
+
+	chks, err := ExpandChunks(encoder.Iterator(nil))
+	require.NoError(t, err)
+	require.Equal(t, test.expectedChunks, len(chks))
+
+	// Decode all encoded samples and assert they are equal to the original ones.
+	encodedSamples := expandHistogramSamples(chks)
+	require.Equal(t, len(test.samples), len(encodedSamples))
+	for i, encodedSample := range encodedSamples {
+		var testSample *histogram.FloatHistogram
+		if test.samples[i].H() != nil {
+			testSample = test.samples[i].H().ToFloat()
+		} else {
```

Comment:
I recommend not converting int histogram to float histogram in the test. Let's test the histograms as is.

Reply:
Since we add a recoding float histogram appender in this PR, should I add separate tests for appending float histograms? The idea behind the conversion was to avoid duplicating tests, especially if we need to add new cases in the future.

Reply:
I guess you already added those test cases now and it looks good, thanks!